data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/v1/beacon/GetBlock.java

```diff
@@ -0,0 +1,89 @@
+import static tech.pegasys.teku.beaconrestapi.RestApiConstants.RES_OK;
+import static tech.pegasys.teku.beaconrestapi.RestApiConstants.TAG_V1_BEACON;
+import static tech.pegasys.teku.infrastructure.http.HttpStatusCodes.SC_BAD_REQUEST;
+import static tech.pegasys.teku.infrastructure.http.HttpStatusCodes.SC_NOT_FOUND;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import io.javalin.http.Context;
+import io.javalin.http.Handler;
+import io.javalin.plugin.openapi.annotations.HttpMethod;
+import io.javalin.plugin.openapi.annotations.OpenApi;
+import io.javalin.plugin.openapi.annotations.OpenApiContent;
+import io.javalin.plugin.openapi.annotations.OpenApiParam;
+import io.javalin.plugin.openapi.annotations.OpenApiResponse;
+import java.util.Map;
+import java.util.Optional;
+import org.jetbrains.annotations.NotNull;
+import tech.pegasys.teku.api.ChainDataProvider;
+import tech.pegasys.teku.api.DataProvider;
+import tech.pegasys.teku.api.response.v1.beacon.GetBlockResponse;
+import tech.pegasys.teku.api.schema.SignedBeaconBlock;
+import tech.pegasys.teku.beaconrestapi.handlers.AbstractHandler;
+import tech.pegasys.teku.beaconrestapi.schema.BadRequest;
+import tech.pegasys.teku.infrastructure.async.SafeFuture;
+import tech.pegasys.teku.provider.JsonProvider;
+
+public class GetBlock extends AbstractHandler implements Handler {
+  public static final String ROUTE = "/eth/v1/beacon/blocks/:block_id";
+  private final ChainDataProvider chainDataProvider;
+
+  public GetBlock(final DataProvider dataProvider, final JsonProvider jsonProvider) {
+    this(dataProvider.getChainDataProvider(), jsonProvider);
+  }
+
+  public GetBlock(final ChainDataProvider chainDataProvider, final JsonProvider jsonProvider) {
+    super(jsonProvider);
+    this.chainDataProvider = chainDataProvider;
+  }
+
+  @OpenApi(
+      path = ROUTE,
+      method = HttpMethod.GET,
+      summary = "Get block",
+      tags = {TAG_V1_BEACON},
+      description = "Retrieves block details for given block id.",
+      pathParams = {@OpenApiParam(name = PARAM_BLOCK_ID, description = PARAM_BLOCK_ID_DESCRIPTION)},
+      responses = {
+        @OpenApiResponse(status = RES_OK, content = @OpenApiContent(from = GetBlockResponse.class)),
+        @OpenApiResponse(status = RES_BAD_REQUEST),
+        @OpenApiResponse(status = RES_NOT_FOUND),
+        @OpenApiResponse(status = RES_INTERNAL_ERROR)
+      })
+  @Override
+  public void handle(@NotNull final Context ctx) throws Exception {
+    final Map<String, String> pathParams = ctx.pathParamMap();
+    try {
+      final SafeFuture<Optional<SignedBeaconBlock>> future =
+          chainDataProvider.getBlock(pathParams.get(PARAM_BLOCK_ID));
+      handleOptionalResult(ctx, future, this::handleResult, SC_NOT_FOUND);
+    } catch (IllegalArgumentException ex) {
+      ctx.status(SC_BAD_REQUEST);
+      ctx.result(BadRequest.badRequest(jsonProvider, ex.getMessage()));
+    }
```

Comment:
this shouldn't be needed? if it's thrown during parse of the block id, that should be caught in the selector, and a BadRequestException should get thrown and it'll get handled by the generic exception handler.

Reply:
Removed.